wire-schema/src/main/java/com/squareup/wire/schema/FileLinker.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema;
+
+import java.util.Collection;
+
+final class FileLinker {
```

Comment:
For follow-up, FileLinker will have phases like `IMPORTS/TYPES/MEMBERS/OPTIONS/VALIDATED` and we’ll only link to the level necessary.

sourcePath will be VALIDATED
direct dependencies will be TYPES
transitive dependencies will be nothing, not even loaded

There’s of course an annoying special case. When we have options we will need to link direct dependencies to the MEMBERS phase, and possibly transitive dependencies also.